apps/core/src/hooks/useNftDetails.ts

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+import {
+    useGetNFTMeta,
+    useOwnedNFT,
+    useNFTBasicData,
+    useGetKioskContents,
+    useIsAssetTransferable,
+} from './';
+import { formatAddress } from '@iota/iota-sdk/utils';
+import { truncateString } from '../utils';
+
+type NftField = { keys: string[]; values: string[] };
+
+type NftFields = {
+    metadata?: { fields?: { attributes?: { fields?: NftField } } };
+};
+
+export function useNftDetails(nftId: string, accountAddress: string | null) {
+    const { data: objectData, isPending: isNftLoading } = useOwnedNFT(nftId || '', accountAddress);
+    const { data } = useGetKioskContents(accountAddress);
+
+    const isContainedInKiosk = data?.lookup.get(nftId!);
+    const kioskItem = data?.list.find((k) => k.data?.objectId === nftId);
+
+    const { data: isTransferable } = useIsAssetTransferable(objectData);
```

Comment:
you can include here `isCheckingAssetTransferability` and either return it or have a grouped isLoading that checks for isNftLoading, isMetaLoading, isCheckingAssetTransferability, etc...

Reply:
Also, now that we have this new hook, I would make use of it everywhere and avoid redundancy, for example:
- https://github.com/iotaledger/iota/blob/78517eefd2f36a837bb6d831677f5b3d437c40c1/apps/wallet/src/ui/app/pages/home/nft-details/index.tsx#L31-L32 can be removed because they could come from the hook
- you could  potentially make use of it in https://github.com/iotaledger/iota/blob/78517eefd2f36a837bb6d831677f5b3d437c40c1/apps/wallet-dashboard/app/(protected)/assets/%5BobjectId%5D/page.tsx#L14 too
- you could potentially make use of it in https://github.com/iotaledger/iota/blob/78517eefd2f36a837bb6d831677f5b3d437c40c1/apps/wallet-dashboard/app/(protected)/assets/%5BobjectId%5D/page.tsx#L14 too